spanner/metrics.go

```diff
@@ -228,18 +277,31 @@ func newBuiltinMetricsTracerFactory(ctx context.Context, dbpath string, metricsP
 		},
 		shutdown: func(ctx context.Context) {},
 	}
-
+	tracerFactory.isAFEMetricsEnabled = strings.EqualFold("false", os.Getenv("SPANNER_DISABLE_AFE_SERVER_TIMING"))
 	tracerFactory.isDirectPathEnabled = false
 	tracerFactory.enabled = false
 	var meterProvider *sdkmetric.MeterProvider
 	if metricsProvider == nil {
 		// Create default meter provider
-		mpOptions, err := builtInMeterProviderOptions(project, compression, opts...)
+		mpOptions, err := builtInMeterProviderOptions(project, compression, tracerFactory.clientAttributes, opts...)
 		if err != nil {
 			return tracerFactory, err
 		}
 		meterProvider = sdkmetric.NewMeterProvider(mpOptions...)
 
+		isEnableGRPCBuiltInMetrics := strings.EqualFold("false", os.Getenv("SPANNER_DISABLE_DIRECT_ACCESS_GRPC_BUILTIN_METRICS"))
+		if isEnableGRPCBuiltInMetrics {
+			mo := opentelemetry.MetricsOptions{
+				MeterProvider: meterProvider,
+				Metrics:       stats.NewMetrics(grpcMetricsToEnable...),
+			}
+			tracerFactory.clientOpts = []option.ClientOption{
+				option.WithGRPCDialOption(
+					opentelemetry.DialOption(opentelemetry.Options{MetricsOptions: mo})),
+				option.WithGRPCDialOption(
+					grpc.WithDefaultCallOptions(grpc.StaticMethodCallOption{})),
```

Comment:
What does this do?

Reply:
Added a comment in code copying here as well 
```
Configure gRPC dial options to enable gRPC metrics collection and static method call option.
The static method call option ensures consistent method names in metrics by preventing gRPC from
automatically adding service prefixes to method names. This helps maintain consistent metric
```